lib/kafka_ex/server.ex

```diff
@@ -281,9 +281,16 @@ defmodule KafkaEx.Server do
             :leader_not_available
           broker -> case produce_request.required_acks do
             0 ->  NetworkClient.send_async_request(broker, produce_request_data)
-            _ -> broker
-            |> NetworkClient.send_sync_request(produce_request_data, state.sync_timeout)
-            |> Produce.parse_response
+            _ ->
+              response = broker
+               |> NetworkClient.send_sync_request(produce_request_data, state.sync_timeout)
+               |> Produce.parse_response
+              case response do
+                [%KafkaEx.Protocol.Produce.Response{partitions: [%{error_code: :no_error, offset: offset, partition: _}], topic: topic}] when offset != nil ->
+                  {:ok, offset}
+                [%KafkaEx.Protocol.Produce.Response{partitions: [%{error_code: 0, offset: offset, partition: _}], topic: topic}] when offset != nil ->
+                  {:ok, offset}
+              end
```

Comment:
Added wildcard clause returning `{:error, response}`.

Yes both success code paths are possible. I've seen `:no_error` in tests, and we see `0` in our service.